src/sentry/grouping/grouping_info.py

```diff
@@ -14,9 +14,9 @@
 
 
 def get_grouping_info(
-    config_name: str | None, project: Project, event_id: str
+    config_name: str | None, project: Project, event_id: str, event: Event | None = None
 ) -> dict[str, dict[str, Any]]:
-    event = eventstore.backend.get_event_by_id(project.id, event_id)
+    event = event or eventstore.backend.get_event_by_id(project.id, event_id)
```

Comment:
nit: what about instead of having event be optional, if we just always passed in the event?

this would mean in the callers, you'd simply make the call to nodestore before calling this function.

reasoning: the interface to this function would be a little less complicated.